packages/design-system/src/Button/Button.styles.tsx

```diff
@@ -0,0 +1,246 @@
+    }
+
+    &:disabled {
+      --button-color-bg: transparent;
+    }
+  `,
+  error: css`
+    --button-color-bg: var(--ads-v2-color-bg-error);
+    --button-color-fg: var(--ads-v2-color-fg-on-error);
+    --button-color-border: transparent;
+
+    &:hover {
+      --button-color-bg: var(--ads-v2-color-bg-error-emphasis);
+      --button-color-fg: var(--ads-v2-color-fg-on-error-emphasis);
+      --button-color-border: transparent;
+    }
+
+    &:active {
+      --button-color-bg: var(--ads-v2-color-bg-error-emphasis-plus);
+      --button-color-fg: var(--ads-v2-color-fg-on-error-emphasis);
+      --button-color-border: transparent;
+    }
+  `,
+};
+
+const LoaderSizes = {
+  sm: css`
+    /* Loading icon size */
+    & > .${ButtonLoadingClassName} > .${ButtonLoadingIconClassName} > svg {
+      width: 12px;
+      height: 12px;
+    }
+  `,
+  md: css`
+    /* Loading icon size */
+    & > .${ButtonLoadingClassName} > .${ButtonLoadingIconClassName} > svg {
+      width: 16px;
+      height: 16px;
+    }
+  `,
+  lg: css`
+    /* Loading icon size */
+    & > .${ButtonLoadingClassName} > .${ButtonLoadingIconClassName} > svg {
+      width: 20px;
+      height: 20px;
+    }
+  `,
+};
+
+export const ButtonContent = styled.div<{
+  size?: Size;
+}>`
+  /* Content is separated out to make opacity driven loader functionality. */
+  /* Size style */
+  ${({ size }) => size && Sizes[size]}
+
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  gap: var(--ads-v2-spaces-2);
+  background-color: var(--button-color-bg);
+  border: 1px solid var(--button-color-border);
+  width: 100%;
+  height: 100%;
+  box-sizing: border-box;
+  padding: var(--button-padding);
+
+  &
+    > .${ButtonContentChildrenClassName},
+    &
+    > .${ButtonContentChildrenClassName}
+    > * {
+    color: var(--button-color-fg);
+    font-family: var(--ads-v2-font-family);
+    font-weight: var(--button-font-weight);
+    font-size: var(--button-font-size);
+  }
+
+  & > .${ButtonContentIconStartClassName} > svg,
+  & > .${ButtonContentIconEndClassName} > svg {
+    width: var(--button-icon-size);
+    height: var(--button-icon-size);
+    color: var(--button-color-fg);
+  }
+`;
+
+export const StyledButton = styled.button<{
+  kind?: Kind;
+  UNSAFE_height?: string;
+  size?: Size;
+  UNSAFE_width?: string;
+  isLoading?: boolean;
+  disabled?: boolean;
+}>`
+  ${Variables}
+
+  /* Variant style */
+  ${({ kind }) => kind && Kinds[kind]}
+
+  /* Loader size style */
+  ${({ size }) => size && LoaderSizes[size]}
+
+  position: relative;
+  cursor: pointer;
+  border-radius: var(--ads-v2-border-radius);
+  border: none;
+  background-color: transparent;
+  ${({ UNSAFE_height }) => UNSAFE_height && `height: ${UNSAFE_height};`}
+  ${({ UNSAFE_width }) => UNSAFE_width && `width: ${UNSAFE_width};`}
+  padding: 0;
+  box-sizing: border-box;
+  overflow: hidden;
+
+  /* button disabled style */
+  ${({ disabled }) =>
```

Comment:
you can simply do:
```
&:disabled {
 cursor: not-allowed;
  opacity: var(--ads-v2-opacity-disabled);
}
```

disabled is already an HTML attribute that you can target with a CSS selector. I try to avoid extracting props in styled components as much as i can as they make code ugly. For e.g for loading state, i do the following:

component:
```
 <StyledButton
      ....
      data-disabled={isDisabled || undefined}
      data-loading={isLoading || undefined}
      disabled={isDisabled || undefined}
      ...
    >
      {content}
    </StyledButton>
```

and in styled component, i target loading state as:

```
&[data-loading] {
 ....
}
```

PS: If you have a doubt about why i used disabled and data-disabled together, DM me, there is a very specific use case for it ( hint: accessibility and polymorphic ).